### PR TITLE
GetBoundary nodes added 2016

### PR DESCRIPTION
**Pull request by moethu**
### Purpose

Added Boundary & Location nodes

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files
- [x] Snapshot of UI changes, if any.

### Reviewers

@mjkkirschner 

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

**Comment by mjkkirschner** [comment]
@kronz is this what you requested at some point?

**Comment by kronz** [comment]
@mjkkirschner @moethu  yes, please

**Comment by mjkkirschner** [comment]
@moethu can you also generate cherry picks for this branch... After we solve the test errors and possible diffs between branches (missing commits)

File: src/Libraries/RevitNodes/Elements/Room.cs [hunk 1]
```diff
@@ -136,6 +136,35 @@ private void InternalSetElement(Autodesk.Revit.DB.Architecture.Room element)
 
         #endregion
 
+        #region Private Helpers
+
+        private IEnumerable<IEnumerable<Autodesk.DesignScript.Geometry.Curve>> GetBoundaries(Autodesk.Revit.DB.SpatialElementBoundaryLocation position)
+        {
+            Autodesk.Revit.DB.SpatialElementBoundaryOptions options = new Autodesk.Revit.DB.SpatialElementBoundaryOptions()
```
Comment (mjkkirschner) [inline src/Libraries/RevitNodes/Elements/Room.cs]: please use Var where possible

File: src/Libraries/RevitNodes/Elements/Room.cs [hunk 2]
```diff
@@ -136,6 +136,35 @@ private void InternalSetElement(Autodesk.Revit.DB.Architecture.Room element)
...
+            {
+                SpatialElementBoundaryLocation = position,
+                StoreFreeBoundaryFaces = true
+            };
+
+            List<List<Autodesk.DesignScript.Geometry.Curve>> boundaries = new List<List<Autodesk.DesignScript.Geometry.Curve>>();
```
Comment (mjkkirschner) [inline src/Libraries/RevitNodes/Elements/Room.cs]: var

File: src/Libraries/RevitNodes/Elements/Room.cs [hunk 3]
```diff
@@ -136,6 +136,35 @@ private void InternalSetElement(Autodesk.Revit.DB.Architecture.Room element)
...
+
+            foreach (List<Autodesk.Revit.DB.BoundarySegment> segments in this.InternalRevitElement.GetBoundarySegments(options))
```
Comment (mjkkirschner) [inline src/Libraries/RevitNodes/Elements/Room.cs]: var

File: src/Libraries/RevitNodes/Elements/Room.cs [hunk 4]
```diff
@@ -136,6 +136,35 @@ private void InternalSetElement(Autodesk.Revit.DB.Architecture.Room element)
...
+            {
+                List<Autodesk.DesignScript.Geometry.Curve> boundary = new List<Autodesk.DesignScript.Geometry.Curve>();
+
+                foreach (Autodesk.Revit.DB.BoundarySegment segment in segments)
```
Comment (mjkkirschner) [inline src/Libraries/RevitNodes/Elements/Room.cs]: var

File: src/Libraries/RevitNodes/Elements/Room.cs [hunk 5]
```diff
@@ -188,6 +217,75 @@ public double Height
...
+        /// <summary>
+        /// Get Room Location
+        /// </summary>
+        public Autodesk.DesignScript.Geometry.Point Location
+        {
+            get
+            {
+                if (this.InternalRevitElement.Location is Autodesk.Revit.DB.LocationPoint)
+                {
+                    Autodesk.Revit.DB.LocationPoint loc = this.InternalRevitElement.Location as Autodesk.Revit.DB.LocationPoint;
```
Comment (mjkkirschner) [inline src/Libraries/RevitNodes/Elements/Room.cs]: var

File: src/Libraries/RevitNodes/Elements/Room.cs [hunk 6]
```diff
@@ -188,6 +217,75 @@ public double Height
...
+                    return loc.Point.ToPoint();
```
Comment (mjkkirschner) [inline src/Libraries/RevitNodes/Elements/Room.cs]: will this scale correctly to host units by default?

File: src/Libraries/RevitNodes/Elements/Room.cs [hunk 7]
```diff
@@ -188,6 +217,75 @@ public double Height
...
+                return null;
+            }
+        }
+
+        /// <summary>
+        /// Check if a point is inside of a room
+        /// </summary>
+        public bool IsInsideRoom(Autodesk.DesignScript.Geometry.Point point)
+        {
+            return this.InternalRevitElement.IsPointInRoom(point.ToRevitType());
```
Comment (mjkkirschner) [inline src/Libraries/RevitNodes/Elements/Room.cs]: same question about host units default?

**Comment by mjkkirschner** [comment]
a few questions

File: src/Libraries/RevitNodes/Elements/Room.cs (hunk 6 above)
Comment (moethu) [inline src/Libraries/RevitNodes/Elements/Room.cs]: yes ToPoint() does so by default

File: src/Libraries/RevitNodes/Elements/Room.cs (hunk 7 above)
Comment (moethu) [inline src/Libraries/RevitNodes/Elements/Room.cs]: same thing, ToPoint(), the ConvertUnits argument is set to default = true

**Comment by mjkkirschner** [comment]
@moethu somehow this branch has conflicts? All the other branch's PRs merged okay. Can you take a look at this?

**Pull request merged**